### DISCUSSION TITLE
No display of data in LiveWire

### DISCUSSION THREAD

**Discussion by ArvinGolshani** [livewire/livewire#8498] [1. Help]
I'm attempting to create a Multi-Step Appointment Booking feature. However, I'm encountering an issue with Livewire and Flowbite.

I have this code in my Blade file for displaying a carousel of dates:

`    @if ($currentStep == 2)

    <div id="controls-carousel" class="w-1/2 mb-5" data-carousel="static">
        <div class="flex mb-5">
            <button type="button"
                class="z-30 flex items-center justify-center h-full cursor-pointer group focus:outline-none"
                data-carousel-prev>
                <span
                    class="inline-flex items-center justify-center w-10 h-10 rounded group-hover:bg-gray-200 dark:group-hover:bg-gray-800/60 group-focus:ring-4 group-focus:ring-white dark:group-focus:ring-gray-800/70 group-focus:outline-none">
                    <svg class="w-4 h-4 text-gray-500 dark:text-gray-800 rtl:rotate-180" aria-hidden="true"
                        xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 6 10">
                        <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                            d="M5 1 1 5l4 4" />
                    </svg>
                    <span class="sr-only">Previous</span>
                </span>
            </button>
            <button type="button"
                class="z-30 flex items-center justify-center h-full cursor-pointer group focus:outline-none"
                data-carousel-next>
                <span
                    class="inline-flex items-center justify-center w-10 h-10 rounded group-hover:bg-gray-200 dark:group-hover:bg-gray-800/60 group-focus:ring-4 group-focus:ring-white dark:group-focus:ring-gray-800/70 group-focus:outline-none">
                    <svg class="w-4 h-4 text-gray-500 dark:text-gray-800 rtl:rotate-180" aria-hidden="true"
                        xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 6 10">
                        <path stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                            d="m1 9 4-4-4-4" />
                    </svg>
                    <span class="sr-only">Next</span>
                </span>
            </button>
        </div>
        <div class="relative overflow-hidden rounded-lg h-20">
            @foreach ($schedules->chunk(7) as $chunk)
                <div class="hidden duration-700 ease-in-out grid grid-cols-7 gap-2" data-carousel-item>
                    @foreach ($chunk as $schedule)
                    <div class="flex flex-col justify-center items-center text-center bg-white rounded-full border cursor-pointer">
                        <div class="text-gray-500">{{ \Carbon\Carbon::parse($schedule->start)->format('D') }}</div>
                        <div class="text-gray-700 font-bold text-2xl">{{ \Carbon\Carbon::parse($schedule->start)->format('d') }}</div>
                    </div>
                    @endforeach
                </div>
            @endforeach
        </div>
    </div>
@endif`

The problem arises when I place this code within @if ($currentStep == 1) and load the page; the carousel displays perfectly without any issues. However, when I place the code within @if ($currentStep == 2), upon opening the page and clicking on the next step button, the function $this->currentStep++ runs in my Livewire component. I can see the step 2 form, but the carousel fails to function properly. I only see the buttons, and the data-carousel-item is not being displayed.

**Comment by JeffGepiga** [comment]
have you tried adding wire:key inside the looped items?

**Reply by ArvinGolshani** [reply]
Yes, I tried. I found that in Livewire, whenever there's a data attribute or JavaScript in a Blade file, when Livewire sends an update request, for example, when I use **wire:model.live** and click on it, the JavaScript and data attributes stop working. I'm not sure if there's a solution for this issue.

**Reply by JeffGepiga** [reply]
have you tried disabling inject_morph_markers in livewire.php ?

**Reply by ArvinGolshani** [reply]
> have you tried disabling inject_morph_markers in livewire.php ?

Yes, I did this too, but it didn't work, you can see the problem in detail in the video I sent below

**Comment by ArvinGolshani** [comment]
https://github.com/livewire/livewire/assets/162874456/d8c910dc-c08e-434d-be9f-a2c082beb5c8

in this video , you can see the problem ... . as you can see in the video , the carousel is not even in @if statment and when livewire sends a update request , it stops working and runs into problem

**Comment by ArvinGolshani** [comment]
Dose any one found any solution ?
